### PR TITLE
Issue #15214: migrated section 4.1.1 Use of optional braces to whole config testing

### PR DESCRIPTION
**Pull request by Zopsss**
#15214 

https://checkstyle.org/google_style.html#a4.1.1

> 4.1.1 Braces are used where optional 	NeedBraces

>needBraces=''{0}'' construct must use '''{}'''s.

### CODE REVIEW

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputUseOfOptionalBraces.java [hunk 1]
```diff
@@ -71,26 +70,25 @@ public void testIf() {
...
     if (condition()) {
       testIf();
     } else // violation ''else' construct must use '{}'s.'
-    testIf();
-    if (condition()) // violation ''if' construct must use '{}'s.'
-    if (condition()) // violation ''if' construct must use '{}'s.'
       testIf();
+    if (condition()) // violation ''if' construct must use '{}'s.'
+      if (condition()) // violation ''if' construct must use '{}'s.'
+          testIf();
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputUseOfOptionalBraces.java:91]: config was complaining about incorrected indented children, config was right here. Formatter didn't intended the children, I guess it got confused by the trailing comments

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputUseOfOptionalBraces.java [hunk 2]
```diff
@@ -117,164 +115,149 @@ public void emptyImplementation() {}
         String.CASE_INSENSITIVE_ORDER.equals("Hello world one!");
       };
   static Runnable r5 = () -> {};
-}
-
-class EmptyBlocks {
-  boolean flag = true;
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputUseOfOptionalBraces.java:123]: moved this class inside main class that's why diff became so huge

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputUseOfOptionalBraces.java [hunk 3]
```diff
@@ -117,164 +115,149 @@ public void emptyImplementation() {}
...
+      if (flag) { /*foo*/ }
       if (flag) {
         /*foo*/
       }
-      if (flag) // violation ''if' construct must use '{}'s.'
-        ;
-      else // violation ''else' construct must use '{}'s.'
-        ;
+      if (flag) ; // violation ''if' construct must use '{}'s.'
+      else ; // violation ''else' construct must use '{}'s.'
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputUseOfOptionalBraces.java:143]: replaced the `;`, code is more readable now

**Review by Zopsss — COMMENTED** [review]
some extra changes:

**Review by romani — APPROVED** [review]
ok to m merge.

in split view and without spaces diff look good.

**Comment by rdiachenko** [comment]
rebased to fix ci pr-age failures

**Pull request merged**